keras_hub/src/models/flux/flux_layers.py

```diff
@@ -0,0 +1,517 @@
+        self.multiplier = 6 if double else 3
+        self.lin = keras.layers.Dense(self.multiplier * dim, use_bias=True)
+
+    def build(self, input_shape):
+        self.lin.build(input_shape)
+
+    def call(self, x):
+        """
+        Generates modulation parameters from the input tensor.
+
+        Args:
+            x: KerasTensor. Input tensor.
+
+        Returns:
+            tuple[ModulationOut, ModulationOut | None]: A tuple containing the shift,
+            scale, and gate tensors. If `double` is True, returns two sets of modulation parameters.
+        """
+        x = keras.layers.Activation("silu")(x)
+        out = self.lin(x)
+        out = keras.ops.split(
+            out[:, None, :], indices_or_sections=self.multiplier, axis=-1
+        )
+
+        return (
+            ModulationOut(*out[:3]),
+            ModulationOut(*out[3:]) if self.is_double else None,
+        )
+
+
+class DoubleStreamBlock(keras.Model):
+    """
+    A block that processes image and text inputs in parallel using
+    self-attention and MLP layers, with modulation.
+
+    Args:
+        hidden_size: int. The hidden dimension size for the model.
+        num_heads: int. The number of attention heads.
+        mlp_ratio: float. The ratio of the MLP hidden dimension to the hidden size.
+        use_bias: bool, optional. Whether to include bias in QKV projection. Default is False.
+    """
+
+    def __init__(
+        self,
+        hidden_size,
+        num_heads,
+        mlp_ratio,
+        use_bias=False,
+    ):
+        super().__init__()
+
+        mlp_hidden_dim = int(hidden_size * mlp_ratio)
+        self.num_heads = num_heads
+        self.hidden_size = hidden_size
+
+        self.image_mod = Modulation(hidden_size, double=True)
+        self.image_norm1 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.image_attn = SelfAttention(
+            dim=hidden_size, num_heads=num_heads, use_bias=use_bias
+        )
+
+        self.image_norm2 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.image_mlp = keras.Sequential(
+            [
+                keras.layers.Dense(mlp_hidden_dim, use_bias=True),
+                keras.layers.Activation("gelu"),
+                keras.layers.Dense(hidden_size, use_bias=True),
+            ]
+        )
+
+        self.text_mod = Modulation(hidden_size, double=True)
+        self.text_norm1 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.text_attn = SelfAttention(
+            dim=hidden_size, num_heads=num_heads, use_bias=use_bias
+        )
+
+        self.text_norm2 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.text_mlp = keras.Sequential(
+            [
+                keras.layers.Dense(mlp_hidden_dim, use_bias=True),
+                keras.layers.Activation("gelu"),
+                keras.layers.Dense(hidden_size, use_bias=True),
+            ]
+        )
+        self.attention = FluxRoPEAttention()
+
+    def call(self, image, text, vec, pe):
```

Comment:
update arg names to be more readable
vec->modulation_vector
pe->positional_encoding